pykmip_demo/README.md

```diff
@@ -0,0 +1,121 @@
+protect it while serving cloud software.
+
+These instructions detail the install and setup processes for Accumulo, pelz, PyKMIP, and a proxy
+server. They proxy server coordinates communication between pelz and PyKMIP. This is because pelz
+does not support TLS endpoints within the Trusted-Exceuction Environment (TEE). The instructions
+then explain how to build Accumulo and run its unit tests using pelz and PyKMIP. The unit tests
+require successful functionality of the PelzCryptoService within Accumulo to protect the file-
+encryption-keys (FEKs) by using the key-encryption-key (KEK) stored within the PyKMIP server. 
+
+## Keys-in-use demonstration process
+
+The process for key protections is outlined here:  
+
+1. Accumulo generates a FEK for its internal file encryption
+2. Accumulo sends the FEK to pelz to be encrypted by the KEK. This request contains the KEK's
+  Unique Identification (UID)
+3. Pelz connects othe proxy server from within the TEE using an ECDH key establishment
+4. The proxy server forwards the key request to the PyKMIP server using TLS
+5. PyKMIP verifies the request and sends the KEK back to the pelz enclave through the proxy server
+6. Pelz uses the KEK within the enclave to encrypt/decrypt FEKs as per Accumulo's requests
+
+## Keys and Certificates
+
+The demostration has four cert/key pairs and two keys for encryption. The two encryption keys are:  
+
+ * The file-encryption-key (FEK)
+ * The key-encryption-key (KEK) 
+
+The KEK is stored on the PyKMIP server. It is used to wrap and unwrap FEKs as requested by Accumulo.
+The FEK is generated by Accumulo for its internal file protections.  
+
+The four certificate/key pairs are:  
+
+ * One certificate for the node
+ * Two for the proxy server
+ * One for PyKMIP server
+
+The proxy server has two certificate/key pairs to protect both the connection to the node and the
+connection to the PyKMIP server.
+
+## Steps for End to End demo of PyKMIP, Accumulo, and pelz. 
+The remainder of this README describes the process for installing the necessary tools to execute
+the demonstration. These instructions omit the installation and setup of an SGX emulator. This must
+be done prior to executing the demo.
+
+### Pelz Installation Steps
+1.  Open terminal
+2.	Clone pelz repo and follow install instructions
+
+		git clone https://github.com/NationalSecurityAgency/pelz.git
+		cd pelz
+
+### Accumulo and Accumulo Plugin Installation Steps 
+3. Download the Accumulo source and setup pelz plugin
+
+		cd ..
+		git clone https://github.com/apache/accumulo.git
+		sudo apt install maven openjdk-11-jdk libxml2-utils
+		cd pelz/accumulo_plugin
+		./setup_plugin.sh -i -d ~/accumulo/ -p
+		cd ..
+
+### PyKMIP Server Installaiton/Setep Steps
+4.  Run PyKMIP Script in a separate terminal
+
+		./pykmip_demo/PyKMIP_setup.sh
+
+5.  Run the server in a separate terminal
+
+		pykmip-server
+
+6.  Register keys with the server
+
+		./pykmip_demo/register_keys_pykmip.sh
```

Comment:
When the PyKMIP database is empty, this script always creates keys with IDs 1-10, but if there are already objects in the database, then the keys will have different IDs.
This is important because the accumulo test always retrieves the object with ID 5.

It would be safer for these instructions to reset the PyKMIP database before running this script to make sure the object with ID 5 is our key.
The best way to clear the database is to delete or move the `/etc/pykmip/pykmip.db` file before starting `pykmip-server`.
That could possibly be added to the `PyKMIP_setup.sh` script.